platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SavedStateMetadata.java

```diff
@@ -289,13 +378,35 @@ private static String parseString(
         return data.get(field);
     }
 
+    /**
+     * Attempt to parse a string from the data map.
+     *
+     * @param data  the data map
+     * @param field the field to parse
+     * @return the parsed string, or null if the field is not present or the value is not a valid hash
+     */
+    @SuppressWarnings("SameParameterValue")
+    @NonNull
+    private static String parseNonNullString(
+            final Map<SavedStateMetadataField, String> data, final SavedStateMetadataField field) throws IOException {
+
+        if (!data.containsKey(field)) {
+            throwMissingRequiredField(field);
+            return null;
```

Comment:
The return value of the method is marked `@NonNull` but this is returning a null.

Reply:
I changed this to the following to avoid the compiler warning. I'm not sure a new exception type would help here... the only way we can avoid the return statement would be to throw the exception directly in the method, as opposed to wrapping it in a helper method (which reduces a lot of duplicated code).

```
if (!data.containsKey(field)) {
    throwMissingRequiredField(field);
    return "we will never reach this point";
}
```